### ISSUE TITLE
json results file is empty @AfterClass

### ISSUE THREAD

**Issue by FaiMal** [cucumber/cucumber-jvm#861]
Hi,

I am using the json results generated by cucumber and converting them into html reports using a 
method generateReports() which is called at @AfterClass. The issue is that the json file is empty when @AfterClass runs. So the html report is empty as well. If I run this method after the execution has completely finished, I see a complete report.

When is the json file populated with results? Is there an @annotation that I can use instead of @AfterClass to perform some action when the json results have been generated

```
@RunWith(Cucumber.class)

@CucumberOptions(features = { "classpath:features/" }, format = { "json:target/JsonResults/cucumber.json" }, dryRun = false)

public class CucumberTestsRun {

 

@BeforeClass

public static void setup() {

    System.out.println("Ran before");

}

 

@AfterClass

public static void teardown() throws Exception {

    System.out.println("Ran after");

    GenerateReports.generateReports();

}

```
Regards,
FS

**Comment by velanbabu** [comment]
I'm also facing same issue. Please provide the solution ASAP.

**Comment by dghighfill** [comment]
I had a similar issue, and I think I've boiled it down to this.  Say you have three feature files and you pass in tags that will only execute tests in one of the features, that is not executed last and the last feature that is looked at is excluded by the tags provided in the cucumber options, then the json file will be empty and the cumber will report zero tests ran.  Here are some snippets from my run.

````-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running com.company.AWDPortalIT
6 Scenarios (6 passed)
35 Steps (35 passed)
2m14.616s

Tests run: 41, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 135.43 sec - in com.company.AWDPortalIT
Running com.company.launchpad.TAProcessingWorkspaceIT
None of the features at [classpath:com/dstsystems/alloy/launchpad] matched the filters: [~@tapw, @awd]

0 Scenarios
0 Steps
0m0.000s

Tests run: 0, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.059 sec - in com.company.launchpad.TAProcessingWorkspaceIT
Results :
Tests run: 41, Failures: 0, Errors: 0, Skipped: 0

[Cucumber Tests] Parsing results.
[Cucumber Tests] parsing cucumber-report.json
No cucumber scenarios appear to have been run.
````

**Comment by bcub** [comment]
I'd the same issue, But I get over it using maven to execute a java application after the cmd **mvn test**
```
<plugin>
<groupId>org.apache.maven.plugins</groupId>
						<artifactId>maven-surefire-plugin</artifactId>
						<version>2.14</version>
						<configuration>
							<testFailureIgnore>true</testFailureIgnore>
						</configuration>
					</plugin>
					<plugin>
						<groupId>org.codehaus.mojo</groupId>
						<artifactId>exec-maven-plugin</artifactId>
						<version>1.1.1</version>
						<executions>
							<execution>
								<phase>test</phase>
								<goals>
									<goal>java</goal>
								</goals>
								<configuration>
									<mainClass>fr.digitas.michelin.bdd.Report</mainClass>
								</configuration>
							</execution>
						</executions>
					</plugin>```

**Comment by dghighfill** [comment]
Just a followup question, you have a class that has a main method that all it does is GenerateReports.generateReports()?  How does that know about all the prior test executions?  We're using the failsafe plugin and running a mvn verify.

**Comment by dkowis** [comment]
Please take this to the cucumber google groups mailing list, unless you've found a legitimate bug in the cucumber-jvm codebase. https://cucumber.io/support

**Comment by dghighfill** [comment]
I believe this is a bug in the reporting of the test results.  If ran with tags and the final feature file looked at is excluded by tags to run any scenarios, the test results write an empty file even though they're may have been tests ran from another test.  Currently my scenario is only with two feature files.  I never noticed this condition with three feature files.  Shall I open a new issue or does this one get reopen?

**Comment by SanjeevKumarmn** [comment]
Is there a solution on this issue? It's a valid point at times we need to auto backup json files after the test execution for better documentation of results.

**Comment by mpkorstanje** [comment]
Should work in 2.0.x as that merged https://github.com/cucumber/cucumber-jvm/pull/1190.

Still what you're trying to do is wrong. Post processing test results should be part of your build, something that happens after the test run has been executed, not integrated in to the test run itself.

**Comment by SanjeevKumarmn** [comment]
Thanks for your response and I'm glad there is a solution. Not all tests are run as CICD or Devops pipeline approach. Sometimes cucumber is used as a regression testing with large number of test cases where developers can't wait for a longer test execution (although tests run in parallel) if we include this part of build. Also for better traceability of results, we backup these json files and generate  HTML results as one single summary of the whole regression suite execution that are executed in multiple machines using a Winium tool(windows application testing using selenium API).
